rust/agents/relayer/src/server.rs

```diff
@@ -0,0 +1,170 @@
+use axum::{
+    extract::{Query, State},
+    routing, Router,
+};
+use derive_new::new;
+use hyperlane_core::H256;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use std::str::FromStr;
+use tokio::sync::broadcast::Sender;
+
+use crate::msg::pending_operation::PendingOperation;
+
+const MESSAGE_RETRY_API_BASE: &str = "/message_retry";
+pub const ENDPOINT_MESSAGES_QUEUE_SIZE: usize = 1_000;
+
+/// Returns a vector of validator-specific endpoint routes to be served.
+/// Can be extended with additional routes and feature flags to enable/disable individually.
+pub fn routes(tx: Sender<MessageRetryRequest>) -> Vec<(&'static str, Router)> {
+    let message_retry_api = MessageRetryApi::new(tx);
+
+    vec![message_retry_api.get_route()]
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum MessageRetryRequest {
+    MessageId(H256),
+    DestinationDomain(u32),
+}
+
+impl PartialEq<Box<dyn PendingOperation>> for &MessageRetryRequest {
+    fn eq(&self, other: &Box<dyn PendingOperation>) -> bool {
+        match self {
+            MessageRetryRequest::MessageId(message_id) => message_id == &other.id(),
+            MessageRetryRequest::DestinationDomain(destination_domain) => {
+                destination_domain == &other.destination_domain().id()
+            }
+        }
+    }
+}
+
+#[derive(new, Clone)]
+pub struct MessageRetryApi {
+    tx: Sender<MessageRetryRequest>,
+}
+
+#[derive(Deserialize)]
+struct RawMessageRetryRequest {
+    message_id: Option<String>,
+    destination_domain: Option<u32>,
+}
+
+impl From<RawMessageRetryRequest> for Vec<MessageRetryRequest> {
```

Comment:
wonder if this should be a TryFrom because it's possible to pass in a malformed message_id, it'd be nice to surface this error in the api response

Reply:
good point, actually we were only returning status codes previously but I added error messages now